src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,6 +183,54 @@ SimpleNameSyntax syntax
 			}
 		}
 
+		private static void AnalyzeConditionalImmutability(
+			SyntaxNodeAnalysisContext ctx,
+			SyntaxNode syntax
+		) {
+			// Ignore references in DocComments such as crefs
+			if( syntax.IsFromDocComment() ) {
+				return;
+			}
+
+			// Methods can be either of the following depending on where they are declared
+			MethodDeclarationSyntax methodSyntax = null;
+			LocalFunctionStatementSyntax functionSyntax = null;
+
+			switch( syntax ) {
+				case MethodDeclarationSyntax:
+					methodSyntax = (MethodDeclarationSyntax)syntax;
+					break;
+				case LocalFunctionStatementSyntax:
+					functionSyntax = (LocalFunctionStatementSyntax)syntax;
+					break;
+				default:
+					return;
+			}
+
+			// Get the parameter list syntax and make sure it is not null
+			TypeParameterListSyntax paramListSyntax = methodSyntax?.TypeParameterList ?? functionSyntax?.TypeParameterList;
+			if( paramListSyntax == null ) {
+				return;
+			}
+
+			// Iterate through the individual parameters of the method
+			foreach( TypeParameterSyntax parameter in paramListSyntax.Parameters ) {
+
+				// Iterate through the attributes on the current parameter
+				foreach( AttributeListSyntax attribute in parameter.AttributeLists ) {
+
+					// Check if the current attribute is the conditional immutability attribute
+					if( attribute.Attributes.ToString() == "ConditionallyImmutable.OnlyIf" ) {

```

Comment:
The downside to using strings to compare symbols is that it's pretty fragile, e.g.

```cs
using static D2L.CodeStyle.Analayzers.Something.ConditionallyImmutable;

// ...

void Foo<[OnlyIf]T>() {}
```

wouldn't get caught. It's not a big deal here but I'll link to an example of how to do this purely with symbols in a sec.

Reply:
I appreciate that, I was expecting you to suggest something like that

Reply:
https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs#L22

You can grab a reference to the symbol from the compilation inside the wireup like this, and pass it down into this method and use it for comparison without conversion to strings.

Reply:
Yeah, it's hard to know what the right thing to do for this stuff is -- we've learned this stuff the hard way haha.

Reply:
I guess reading a lot of documentation would help but (A) who does that (B) it is quite scattered/non-existent.